### PR TITLE
Fix game pools with __SANITIZE_ADDRESS__

### PR DESCRIPTION
**Pull request by xezon**
This change fixes game pools with `__SANITIZE_ADDRESS__`.

Before, textures would just render black in w3dview tool. Unclear why exactly, but this change reinstates game pools and replaces the pool allocator with `malloc` and `free`.

Textures render normally now.

### CODE REVIEW

File: src/game/common/system/mempool.cpp [hunk 1]
```diff
@@ -139,7 +143,9 @@ void MemoryPool::Free_Block(void *block)
     if (block == nullptr) {
         return;
     }
-
+#ifdef __SANITIZE_ADDRESS__
+    free(block);
```
Comment (feliwir) [inline src/game/common/system/mempool.cpp:147]: I think we should return afterwards? otherwise it will try deleting it otherwise / applying "delete"

Comment (xezon) [reply src/game/common/system/mempool.cpp:147]: Where is that delete? I do not see that.

File: src/game/common/system/gamememory.cpp [hunk 2]
```diff
@@ -94,8 +94,6 @@ void Init_Memory_Manager()
         captainslog_dbgassert(g_thePreMainInitFlag, "memory manager is already inited");
     }
 
-#if defined GAME_DEBUG && !defined __SANITIZE_ADDRESS__
```
Comment (xezon) [inline src/game/common/system/gamememory.cpp:97]: This looks wrong.

Comment (xezon) [reply src/game/common/system/gamememory.cpp:97]: Fixed.

**Review by feliwir — APPROVED** [review]

File: src/game/common/system/gamememory.cpp (hunk 2 above)
Comment (Suardi1985) [reply src/game/common/system/gamememory.cpp:97]: Wtf

**Review by Suardi1985 — APPROVED** [review]

**Review by Suardi1985 — APPROVED** [review]

**Review by Suardi1985 — APPROVED** [review]

**Review by Suardi1985 — APPROVED** [review]

**Pull request merged**